curvefs/src/client/s3/disk_cache_manager_impl.cpp

```diff
@@ -56,13 +56,45 @@ int DiskCacheManagerImpl::Init(const S3ClientAdaptorOption option) {
 }
 
 void DiskCacheManagerImpl::Enqueue(
-  std::shared_ptr<PutObjectAsyncContext> context) {
+    std::shared_ptr<PutObjectAsyncContext> context, bool isReadCacheOnly) {
+    if ( isReadCacheOnly ) {
+        auto task = [this, context]() {
+            this->WriteReadDirectClosure(context);
+        };
+        taskPool_.Enqueue(task);
+        return;
+    }
     auto task = [this, context]() {
         this->WriteClosure(context);
     };
     taskPool_.Enqueue(task);
 }
 
+
+
+int DiskCacheManagerImpl::WriteReadDirectClosure(
+    std::shared_ptr<PutObjectAsyncContext> context) {
+        VLOG(9) << "WriteReadClosure start, name: " << context->key;
+        // Upload to s3
+        int s3Ret = client_->Upload(context->key,
```

Comment:
synchronous upload may affects performance, maybe we can put  the enqueue(WriteReadDirect)  to PutObjectAsyncCallBack  which is defined in DataCache::Flush？

and i think we don't need to care if the cache read disk is successfully written

Reply:
yeah，I will do this